tests/bluetooth/bsim_bt/bsim_test_advx/src/main.c

```diff
@@ -921,6 +929,39 @@ static void test_scanx_main(void)
 		k_sleep(K_MSEC(100));
 	}
 
+	printk("Start scanning for a duration...");
+	is_scan_timeout = false;
+	scan_param.interval = 0x08;
+	scan_param.timeout = 100;
+	err = bt_le_scan_start(&scan_param, scan_cb);
+	if (err) {
+		goto exit;
+	}
+	printk("success.\n");
+
+	printk("Waiting...");
+	while (!is_scan_timeout) {
```

Comment:
I think it would be better to wait for the set duration and then check the value of is_scan_timeout to avoid hanging if the termination fails.

Reply:
ok.. I will use k_sleep for 1010 ms and then check if `is_scan_timeout` has been set.